File: invest-sns/scripts/check_missing_prices.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
import requests, json

key = open('.env.local').readlines()[2].split('=',1)[1].strip()
h = {'apikey': key, 'Authorization': 'Bearer ' + key}
r = requests.get('https://arypzhotxflimroprmdk.supabase.co/rest/v1/influencer_signals?select=id,stock,ticker,signal', headers=h)
signals = r.json()

# All unique tickers from DB
db_tickers = {}
null_tickers = []
for s in signals:
    if s['ticker'] and s['ticker'].strip():
        if s['ticker'] not in db_tickers:
            db_tickers[s['ticker']] = s['stock']
    else:
        null_tickers.append(s)

# Load stockPrices.json
sp = json.load(open('data/stockPrices.json', 'r', encoding='utf-8'))
# Load signal_prices.json
sigp = json.load(open('data/signal_prices.json', 'r', encoding='utf-8'))

print(f"=== DB 시그널 현황 ===")
print(f"Total signals: {len(signals)}")
print(f"Unique tickers: {len(db_tickers)}")
print(f"Null ticker signals: {len(null_tickers)}")

print(f"\n=== stockPrices.json ===")
print(f"Total entries: {len(sp)}")
has_prices = [k for k,v in sp.items() if v.get('prices') and len(v['prices']) > 0]
no_prices = [k for k,v in sp.items() if not v.get('prices') or len(v['prices']) == 0]
print(f"With price history: {len(has_prices)}")
print(f"Without price history: {len(no_prices)}")
if no_prices:
    print(f"  Missing prices: {no_prices}")

print(f"\n=== ticker 있는데 stockPrices.json에 없는 종목 ===")
missing_sp = []
for ticker, stock in sorted(db_tickers.items()):
    if ticker not in sp:
        in_sigp = '✅' if stock in sigp or ticker in sigp else '❌'
        missing_sp.append((ticker, stock, in_sigp))
        print(f"  {ticker} ({stock}) | signal_prices: {in_sigp}")
print(f"Total missing from stockPrices: {len(missing_sp)}")

print(f"\n=== signal_prices.json에서 price null인 종목 ===")
for name, v in sorted(sigp.items()):
    if v.get('current_price') is None:
        print(f"  {name} | ticker={v.get('ticker','?')}")

print(f"\n=== ticker가 null인 시그널 ===")
for s in null_tickers:
    print(f"  [{s['id'][:8]}] {s['stock']} | signal={s['signal']}")

# Classify missing tickers
print(f"\n=== 분류 ===")
coins = ['BTC', 'ETH', 'DOGE', 'SOL', 'KLAY']
kr_pattern = lambda t: t.isdigit() and len(t) == 6
for ticker, stock, _ in missing_sp:
    if ticker in coins:
        cat = 'COIN'
    elif kr_pattern(ticker):
        cat = 'KR'
    else:
        cat = 'US'
    print(f"  {cat}: {ticker} ({stock})")

File: invest-sns/scripts/check_missing_prices.js
const {createClient} = require('@supabase/supabase-js');
const fs = require('fs');

const sb = createClient(
  'https://arypzhotxflimroprmdk.supabase.co',
  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImFyeXB6aG90eGZsaW1yb3BybWRrIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NzIwMDYxMTAsImV4cCI6MjA4NzU4MjExMH0.qcqFIvYRiixwu609Wjj9H3HxscU8vNpo9nS_KQ3f00A'
);

(async()=>{
  const {data,error} = await sb.from('influencer_signals').select('id,stock,ticker,created_at');
  if(error){console.log('ERROR',error);return;}
  console.log('Total signals:', data.length);
  
  const prices = JSON.parse(fs.readFileSync('data/signal_prices.json','utf8'));
  const covered = new Set(Object.keys(prices));
  console.log('Covered:', covered.size);
  
  const missing = data.filter(d=>!covered.has(d.id));
  console.log('Missing:', missing.length);
  missing.forEach(m=>console.log('  '+m.id.slice(0,8)+'.. | '+m.stock+' | '+m.ticker+' | '+m.created_at.slice(0,10)));
  
  // Also check covered ones that might have issues (no signal_price or current_price = 0)
  let badData = 0;
  for(const [id, p] of Object.entries(prices)) {
    if(!p.signal_price || p.signal_price === 0) badData++;
  }
  console.log('\nCovered but signal_price=0 or null:', badData);
  
  // unique tickers
  const tickers = {};
  data.forEach(d=>{if(d.ticker) tickers[d.ticker]=d.stock;});
  console.log('\nUnique tickers ('+Object.keys(tickers).length+'):');
  Object.keys(tickers).sort().forEach(t=>console.log('  '+t+' -> '+tickers[t]));
})();
